### PR TITLE
Update SearchModel when CustomNodeInfo has changed

### PR DESCRIPTION
**Pull request by pboyer**
### Issue

A few regression tests focusing on the side effects of Custom Node Save As were failing.  I discovered that the `SearchModel` was not being updated after performing this operation, although the `CustomNodeSearchElement` was being updated.  The result is the element could not be searched for, although you could find it by browsing.

### Fix 

I inserted a `SearchModel.Update` call on the `CustomNodeSearchElement.`  This basically just removes and re-inserts the `SearchElement`.

### Tests

This fix *removes* some redundant/outdated tests and fixes the existing one.  

### Reviewer

- [ ] @lukechurch

### CODE REVIEW

File: test/DynamoCoreTests/WorkspaceSaving.cs [hunk 1]
```diff
@@ -659,49 +604,6 @@ public void CustomNodeSaveAsNewCustomNodeCanBeUsedInWorkspaceAndExpressionRuns()
...
-            Assert.IsAssignableFrom(typeof(CustomNodeSearchElementViewModel), res1);
-            Assert.IsAssignableFrom(typeof(CustomNodeSearchElementViewModel), res2);
-
-            var node1 = res1.Model as CustomNodeSearchElement;
-            var node2 = res2.Model as CustomNodeSearchElement;
-
-            Assert.IsTrue((node1.ID == oldId && node2.ID == newId) ||
-                          (node1.ID == newId && node2.ID == oldId));
-
-        }
```
Comment (pboyer) [inline test/DynamoCoreTests/WorkspaceSaving.cs]: This basically duplicates the test below.

File: test/DynamoCoreTests/WorkspaceSaving.cs [hunk 2]
```diff
@@ -593,30 +562,6 @@ public void CustomNodeSaveAsAddsNewLoadedNodeToCustomNodeManager()
...
-
-            Assert.IsNotNull(nodeWorkspace);
-            var oldId = nodeWorkspace.CustomNodeDefinition.FunctionId;
-
-            var newPath = GetNewFileNameOnTempPath("dyf");
-            nodeWorkspace.SaveAs(newPath, ViewModel.Model.EngineController.LiveRunnerCore);
-
-            var newDef = nodeWorkspace.CustomNodeDefinition;
-        }
-
```
Comment (pboyer) [inline test/DynamoCoreTests/WorkspaceSaving.cs]: This basically does no assertions!

**Comment by ramramps** [comment]
@pboyer   LGTM.

File: test/DynamoCoreTests/WorkspaceSaving.cs [hunk 3]
```diff
@@ -778,83 +684,26 @@ public void CustomNodeSaveAsAddsNewCustomNodeToSearchAndItCanBeRefactoredWhilePr
...
-            var node1 = (CustomNodeSearchElement)res1.Model;
-            var node2 = (CustomNodeSearchElement)res2.Model;
-
-            Assert.IsTrue((node1.ID == oldId && node2.ID == newId) ||
-                          (node1.ID == newId && node2.ID == oldId));
+            Assert.AreEqual(1, searchResult2.Count);
+            var node4 = searchResult2[0] as CustomNodeSearchElement;
+            Assert.NotNull(node4);
+            Assert.AreEqual(newId, node4.ID);
```
Comment (pboyer) [inline test/DynamoCoreTests/WorkspaceSaving.cs]: This test attempts to test something that's currently not allowable in Dynamo.  That is, users cannot set a custom node name directly without a Save as.  Hence, I don't really see a reason to keep this test around.

**Pull request closed without merging**